spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverLoggingAutoConfiguration.java

```diff
@@ -29,23 +29,22 @@
 import org.springframework.cloud.gcp.autoconfigure.logging.extractors.XCloudTraceIdExtractor;
 import org.springframework.cloud.gcp.autoconfigure.logging.extractors.ZipkinTraceIdExtractor;
 import org.springframework.cloud.gcp.autoconfigure.trace.StackdriverTraceAutoConfiguration;
-import org.springframework.cloud.sleuth.autoconfig.SleuthProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
 
 /**
  * This class configures a Web MVC interceptor to capture trace IDs for log correlation.
- * This configuration is turned on only if Sleuth is not used and Web MVC is used.
+ * This configuration is turned on only if Trace support is not used and Web MVC is used.
  * Otherwise, the MDC context will be used by the Logback appenders.
  *
  * @author Mike Eltsufin
  * @author Chengyuan Zhao
  */
 @Configuration
 @ConditionalOnClass({HandlerInterceptorAdapter.class, LoggingAppender.class})
-@ConditionalOnMissingBean(SleuthProperties.class)
+@ConditionalOnMissingBean(StackdriverTraceAutoConfiguration.class)
```

Comment:
Well, there is this property on the `@ConditionalOnMissingBean` instead:
```
/**
	 * The class type names of bean that should be checked. The condition matches when
	 * each class specified is missing in the {@link ApplicationContext}.
	 * @return the class type names of beans to check
	 */
	String[] type() default {};
```
Exactly for use-case like yours.

WDYT?